Source/RapyutaSimulationPlugins/Private/Core/RRGameMode.cpp

```diff
@@ -0,0 +1,214 @@
+    UE_LOG(LogRapyutaCore, Display, TEXT("- bBenchmark: %d"), bBenchmark);
+    UE_LOG(LogRapyutaCore, Display, TEXT("- SimType: %s"), *GetSimTypeName());
+    UE_LOG(LogRapyutaCore, Display, TEXT("- CaptureResolution: %s"), *CaptureResolution.ToString());
+}
+
+void ARRGameMode::PrintUEPreprocessors()
+{
+    UE_LOG(LogRapyutaCore, Display, TEXT("[ARRGameMode]: UE PREPROCESSORS:"));
+    UE_LOG(LogRapyutaCore, Display, TEXT("* [DO_CHECK] %s!"), URRCoreUtils::GetBoolPreprocessorText<DO_CHECK>());
+
+    UE_LOG(LogRapyutaCore, Display, TEXT("* [STATS] %s!"), URRCoreUtils::GetBoolPreprocessorText<STATS>());
+
+    UE_LOG(LogRapyutaCore,
+           Display,
+           TEXT("* [CPUPROFILERTRACE_ENABLED] (for UE Insights) %s!"),
+           URRCoreUtils::GetBoolPreprocessorText<CPUPROFILERTRACE_ENABLED>());
+
+    UE_LOG(LogRapyutaCore, Display, TEXT("* [WITH_EDITOR] %s!"), URRCoreUtils::GetBoolPreprocessorText<WITH_EDITOR>());
+
+    UE_LOG(LogRapyutaCore,
+           Display,
+           TEXT("* [RHI_RAYTRACING] %s %d!"),
+           URRCoreUtils::GetBoolPreprocessorText<RHI_RAYTRACING>(),
+           IsRayTracingEnabled());
+
+    UE_LOG(LogRapyutaCore, Display, TEXT("* [WITH_UNREALPNG] %s!"), URRCoreUtils::GetBoolPreprocessorText<WITH_UNREALPNG>());
+
+    UE_LOG(LogRapyutaCore, Display, TEXT("* [WITH_UNREALJPEG] %s!"), URRCoreUtils::GetBoolPreprocessorText<WITH_UNREALJPEG>());
+
+#if (!WITH_EDITOR)
+    UE_LOG(LogRapyutaCore,
+           Display,
+           TEXT("- bAsyncLoadingThreadEnabled: %d"),
+           FAsyncLoadingThreadSettings::Get().bAsyncLoadingThreadEnabled);
+#endif
+
+    // Fetch [-physxDispatcher]
+    // Ref: UnrealEngine/Engine/Source/Runtime/Engine/Private/PhysicsEngine/PhysScene_PhysX.cpp:1812
+    int8 physXDispatcherNum = 0;
+    if (PhysSingleThreadedMode())
+    {
+        // UnrealEngine/Engine/Source/Runtime/Engine/Private/PhysicsEngine/PhysScene_PhysX.cpp:1519
+        UE_LOG(LogTemp, Display, TEXT("PHYSICS RUNS IN GAME THREAD!"));
+    }
+    else if (URRCoreUtils::GetCommandLineArgumentValue<int8>(URRCoreUtils::CCMDLINE_ARG_INT_PHYSX_DISPATCHER_NUM,
+                                                             physXDispatcherNum))
+    {
+        UE_LOG(LogTemp, Display, TEXT("PHYSICS RUNS IN [%d] THREADS!"), physXDispatcherNum);
+    }
+    else
+    {
+        UE_LOG(LogTemp, Display, TEXT("PHYSICS RUNS IN A SINGLE THREAD!"));
+    }
+}
+
+void ARRGameMode::ConfigureSimInPlay()
+{
+    if (CaptureResolution.Size() > 0)
+    {
+        UGameUserSettings* gameSettings = GEngine->GetGameUserSettings();
+        gameSettings->SetFullscreenMode(EWindowMode::Windowed);
+        gameSettings->SetScreenResolution(FIntPoint(CaptureResolution.X, CaptureResolution.Y));
+        gameSettings->ApplyResolutionSettings(false);
+    }
+
+    URRCoreUtils::ExecuteConsoleCommand(this, URRCoreUtils::CMD_RHI_ENABLE_GPU_CAPTURE_OPTIONS);
```

Comment:
Yeah, so that is the UE built-in console command, not the OS provided console, typically used to quickly configure graphics options. This like above is part of synth data generation.

Reply:
Right, in terms of API friendliness, UE should not trap users into crash by some mis-configured value and issue use a Warning log instead. Generally, since UE has numerous configs related to GPU (like for ray-tracing), it should properly handle those config values by RHI as in line with the GPU status.
But yeah for sure, let me move this to `RRInternalGameMode` since it is part of scene capturing. Btw, the handling callback for this configure is [here](https://github.com/EpicGames/UnrealEngine/blob/release/Engine/Source/Runtime/RHI/Private/DynamicRHI.cpp#L380).